library/std/src/sync/mutex.rs

```diff
@@ -183,8 +183,33 @@ pub struct Mutex<T: ?Sized> {
 
 // these are the only places where `T: Send` matters; all other
 // functionality works fine on a single thread.
+
+/// SAFETY
+///
+/// The `Send` and `Sync` implementations for `Mutex` ensure that it is safe to
+/// share instances of `Mutex` between threads when the protected data is also
+/// thread-safe. The following explains the safety guarantees:
+///
+/// - `Send` is implemented for `Mutex<T>` if and only if `T` is also `Send`.
+///   This guarantees that `Mutex<T>` can be safely transferred between threads,
+///   and `T` can be sent across thread boundaries. This is crucial for allowing
+///   safe access to the protected data from multiple threads.
```

Comment:
I think this statement is obvious, because `Send` doc states:

> Types that can be transferred across thread boundaries.

I'd much prefer to see exactly **why** is it crucial for `T` to be `Send`. So I think this should be changed to something along the lines of:

> `Mutex` is container which wraps `T`, so it's necessary for `T` to be `Send` to safely send `Mutex` to another thread.

Reply:
alright